client/src/sections/courses/LearningSection/AddLearnerModal.tsx

```
@@ -1,125 +1,162 @@
+  const router = useRouter();
+  const params = router.query;
+  const [search, setSearch] = useState('');
+  const [page, setPage] = useState(1);
+  const [selectedIds, setSelectedIds] = useState<number[]>([]);
+  const courseID = params.id;
+  const [enrollTrainees] = useEnrollLearnerMutation();
+  const { data: trainee } = useGetLearnerQuery({
+    courseId: courseID,
+    isEnrolled: false,
+    pageNumber: page,
+    searchQuery: search,
+  });
+  const enrollees = trainee?.learners.data;
 
-  const handleModalClose = (): void => {
-    setIsModalOpen(false);
+  const handleAddMaterialModal = (): void => {
+    closeModal();
   };
 
   const handleSearch = (search: string): void => {
-    alert(`You searched ${search}`);
+    setSearch(search);
+    setPage(1);
   };
 
-  const learnersPerPage = 8;
-  const totalPages = Math.ceil(learners.length / learnersPerPage);
+  const handleChangePageEvent = (page: number): void => {
+    setPage(page);
+  };
 
-  const lastIndex = currentPage * learnersPerPage;
-  const firstIndex = lastIndex - learnersPerPage;
-  const currentLearners = learners.slice(firstIndex, lastIndex);
+  const handleCheckboxChange = (id: number): void => {
+    if (selectedIds.includes(id)) {
+      setSelectedIds(selectedIds.filter((selectedId) => selectedId !== id));
+    } else {
+      setSelectedIds([...selectedIds, id]);
+    }
+  };
 
-  const handleChangePageEvent = (page: number): void => {
-    setCurrentPage(page);
+  const handleAddLearners = async (): Promise<void> => {
+    const postData = {
+      trainee: selectedIds.join(','),
+    };
+    try {
+      const res = await enrollTrainees({
+        courseId: courseID,
+        postData,
+      });
+      handleAddMaterialModal();
+      if ('data' in res) {
+        alertSuccess(res.data.message);
+      }
+    } catch (error) {
+      alertError('An error occurred while enrolling learners');
+    }
```

I think the error line is not working as expected, please also take server side errors into consideration, if response returns an error, please throw an error so your line 67 can catch it and show toast message properly

---

ill just remove it since, its not doing anything at all